biochemist-python/chapters/Working_with_Pandas.ipynb

```diff
@@ -0,0 +1,1172 @@
+    "\n",
+    "### 1D arrays\n",
+    "A one dimensional array is simply a list of items, for example, a list of the elements: H, He, Li, Be, B, etc.\n",
+    "\n",
+    "### 2D arrays\n",
+    "A two dimensional array contains matched pairs of information. A list of elements with their atomic numbers would be a 2D array:  (H, 1), (He, 2), (Li, 3), (Be, 4), (B, 5), etc. It could also be a list of x, y data points that you want to plot: (1,1), (2,3), (3,5), (4,7). \n",
+    "\n",
+    "### 3D arrays\n",
+    "A three dimensional array contains matched triads of information. One common 3D array in biochemistry and structural biology would be a collection of x, y, and z coordinates for the atoms in a protein, as are found in the [Protein Data Bank](http://www.rcsb.org/pdb). \n",
+    "\n",
+    "The numpy library has functions that will manage n-dimensional arrays, while pandas only works on 2D arrays. You may need numpy for nD arrays at some point, but for this workshop, we will learn to use pandas as we build toward the ability to perform linear and nonlinear regression of laboratory data based on data in 2D arrays.\n",
+    "\n",
+    "The pandas library contains powerful tools for working with 2D data arrays, including the ability to identify the rows and columns of data by unique identifiers: things like \"mass protein\" and \"initial velocity\" rather than \"row 1\" or \"column C\"). Pandas also has many more functions that we will not explore in this workshop, but here are two excellent free online resources for learning more about pandas:\n",
+    "\n",
+    "+ Charlie Weiss's excellent online textbook, *Scientific Computing for Chemists*, which you can find on his GitHub site, [SciCompforChemists](https://github.com/weisscharlesj/SciCompforChemists)\n",
+    "+ Corey Schafer's [Pandas Tutorials on YouTube](https://www.youtube.com/watch?v=ZyhVh-qRZPA&list=PL-osiE80TeTsWmV9i9c58mdDCSskIFdDS)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importing python libraries\n",
+    "\n",
+    "As we move toward data analysis, we will need to import a series python **libraries**. A **library** is a set of modules which contain related functions that can be used to complete specific tasks. Using libraries in Python reduces the amount of code you have to write. Usually a function has some type of input and gives a particular output.  To use a function that is in a library, you often use the dot notation introduced in the previous lesson.\n",
+    "\n",
+    "In the last lesson, we imported the `os` library, which makes it easy to assign the location of a file to a single variable (e.g. datafile), rather than having to input the full path every time (e.g., Users/user_name/Desktop/python-scripting-biochemistry/biochemist-python/chapters/data). In this lesson, we will be using the `numpy` and `pandas`, and we will add a few more libraries in the future. As we proceed, we'll build this table. \n",
```

Comment:
The `os` library was used to construct a file path to avoid differences in file paths on different operating systems. We used it to construct a relative file path, but we could have built a full file path as well. The primary purpose of `os` was to make it so our script could be transferred between computers with different operating systems (mac vs windows)

Reply:
I modified the text to read, "In the last lesson, we imported the `os` library, which can be used to assign the location of a file to a variable (e.g. datafile), so our script could be transferred between computers with different operating systems (e.g., mac vs windows). "